vl-proza-message.src.js

```diff
@@ -136,6 +171,92 @@ export class VlProzaMessage extends VlElement(HTMLElement) {
             this._element.appendChild(this._getEditButtonTemplate());
         }
     }
+
+    __initWysiwyg() {
+        this.__unwrapWysiwygElement();
+        tinyMCE.baseURL = '/node_modules/tinymce';
+        tinyMCE.init(this._wysiwygConfig);
+        this._activeWysiwygEditor.on('init', () => {
+            this.__focusWysiwyg();
+            this.__configureWysiwygStyle();
+            this.__hideWysiwygButton();
+            this.__bindWysiwygEvents();
+        });
+    }
+
+    __focusWysiwyg() {
+        const editor = this._activeWysiwygEditor;
+        editor.focus();
+        editor.selection.select(editor.getBody(), true);
+        editor.selection.collapse(false);
+    }
+
+    __configureWysiwygStyle() {
+        this._activeWysiwygEditor.bodyElement.classList.add('vl-typography');
+    }
+
+    __bindWysiwygEvents() {
+        const editor = this._activeWysiwygEditor;
+        editor.on('keydown', (e) => this.__processKeydownEvent(e));
+        editor.on('blur', (e) => this.__processBlurEvent(e));
+    }
+
+    __processKeydownEvent(e) {
+        const editor = this._activeWysiwygEditor;
+        if (e.keyCode === 27) {
+            while (editor.undoManager.hasUndo()) {
+                editor.undoManager.undo();
+            }
+            this.__stopWysiwyg(editor);
+        }
+        if (e.keyCode === 13 && !e.shiftKey) {
+            editor.undoManager.undo();
+            this.__stopWysiwyg(editor);
+        }
```

Comment:
Ik zou de keycodes nog uit constanten halen of in commentaar er op zijn minst bijzetten over wat voor keys het gaat bij de if-testen.
Misschien ook best ergens documenteren welke keys wat doen voor gebruikers van deze webcomponent en/of een help toets voorzien bij de editor die dit beschrijft.

Reply:
En ik vind het misschien toch nog te uitgebreid per if-test. Het is misschien makkelijker te lezen als er stond:

`
        if (e.keyCode === 27) {
            this._undoAll();
        }
        if (e.keyCode === 13 && !e.shiftKey) {
            this._undo();
        }
`